src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanQuickFix.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2023 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Hani Damlaj
+ *     IBM Corporation
+ *******************************************************************************/
+
+package io.openliberty.tools.intellij.lsp4jakarta.lsp4ij.cdi;
+
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.util.PsiTreeUtil;
+import io.openliberty.tools.intellij.lsp4jakarta.lsp4ij.codeAction.proposal.quickfix.InsertAnnotationMissingQuickFix;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.codeaction.JavaCodeActionContext;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.corrections.proposal.ChangeCorrectionProposal;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.corrections.proposal.ReplaceAnnotationProposal;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.openliberty.tools.intellij.lsp4jakarta.lsp4ij.cdi.ManagedBeanConstants.SCOPE_FQ_NAMES;
+
+public class ManagedBeanQuickFix extends InsertAnnotationMissingQuickFix {
+    public ManagedBeanQuickFix() {
+        super("jakarta.enterprise.context.Dependent");
+    }
+
+    private static final String[] REMOVE_ANNOTATION_NAMES = new ArrayList<>(SCOPE_FQ_NAMES).toArray(new String[SCOPE_FQ_NAMES.size()]);
+
+    @Override
+    protected void insertAnnotations(Diagnostic diagnostic, JavaCodeActionContext context,
+                                     List<CodeAction> codeActions) {
+        String[] annotations = getAnnotations();
+        for (String annotation : annotations) {
+            insertAndReplaceAnnotation(diagnostic, context, codeActions, annotation);
```

Comment:
Seeing this loop here is misleading as it looks like annotations.length == 1 all the time, but if it ever becomes possible that this generates multiple code actions, the context needs to be copied somewhere.

Reply:
Yes, you're right. I think we could just replace it with `context.copy()`. Unfortunately I've seen this code block many times among these quick fixes so I'll go look for other instances.

Reply:
No other instances in this pull request.